### PR TITLE
Add test helper rotation tool

### PR DESCRIPTION
**Pull request by FedericoCeratto**

### CODE REVIEW

File: af/analysis/rotation.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import sys
+import time
+
+import psycopg2  # debdeps: python3-psycopg2
+import statsd  # debdeps: python3-statsd
+import digitalocean  # debdeps: python3-digitalocean
+
+metrics = statsd.StatsClient("127.0.0.1", 8125, prefix="rotation")
+
+log = logging.getLogger("reprocessor")
```
Comment (bassosimone) [inline af/analysis/rotation.py]:
```suggestion
log = logging.getLogger("rotation")
```

File: af/analysis/rotation.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    droplets = api.get_all_droplets(tag_name=TAG)
+    for d in droplets:
+        assert TAG in d.tags
+    live_droplets = [d for d in droplets if d.status == "active"]
+    log.info(f"{len(droplets)} droplets")
+    log.info(f"{len(live_droplets)} active droplets")
+
+    # Avoid failure modes where we destroy all VMs or create unlimited amounts
+    # or churn too quickly
+    # TODO: age cehck
```
Comment (bassosimone) [inline af/analysis/rotation.py]:
```suggestion
    # TODO: age check
```

**Review by bassosimone — APPROVED** [review]
Thanks, I like this!

🐳

File: af/analysis/rotation.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        droplet.destroy()
+
+
+@metrics.timer("spawn_new_droplet")
+def spawn_new_droplet(api, dig_oc_token, live_regions, conf):
+    regions = set(r.slug for r in api.get_all_regions() if r.available is True)
+    preferred_regions = regions - live_regions
+    region = random.choice(list(preferred_regions or regions))
+
+    name = datetime.utcnow().strftime("roaming-th-%Y%m%d%H%M%S")
```
Comment (hellais) [inline af/analysis/rotation.py:134]: Is there a chance for a race on the name of the droplet? Maybe for extra safety you can append a random hash to the end of it?

File: af/analysis/rotation.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    draining_at timestamp without time zone
+);
+ALTER TABLE test_helper_instances OWNER TO shovel;
+
+Example for /etc/ooni/rotation.conf
+--
+[DEFAULT]
+token = CHANGEME
+db_uri = postgresql://shovel:CHANGEME@localhost/metadb
+active_droplets_count = 4
```
Comment (hellais) [inline af/analysis/rotation.py:25]: Shouldn't there be in the settings also `draining_time_minutes`?

File: af/analysis/rotation.py (hunk 1 above)
Comment (FedericoCeratto) [reply af/analysis/rotation.py]: Done

File: af/analysis/rotation.py (hunk 2 above)
Comment (FedericoCeratto) [reply af/analysis/rotation.py]: Fixed

File: af/analysis/rotation.py (hunk 3 above)
Comment (FedericoCeratto) [reply af/analysis/rotation.py:134]: No, starting a new droplet takes much longer than a second

File: af/analysis/rotation.py (hunk 4 above)
Comment (FedericoCeratto) [reply af/analysis/rotation.py:25]: Added

**Review by hellais — APPROVED** [review]
This looks good to merge. Maybe it's helpful to link in the docstring to the specific setup we are using for deploying this in `ooni/sysadmin` pointing to the files `rotation.conf` and `rotation_setup.sh` so it's easy to find.

Beyond that the logic looks reasonable and all the questions I had have been resolved. So it looks good to me.

**Pull request merged**